gendoc/xmake.lua

```diff
@@ -0,0 +1,31 @@
+--!A cross-platform build utility based on Lua
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- Copyright (C) 2015-present, TBOOX Open Source Group.
+--
+-- @author      charlesseizilles
+-- @file        gendoc.lua
+--
+
+task("gendoc")
```

Comment:
If we're just building the document as an html page, we don't really need to provide it as an xmake plugin. Because it is not used by the user. The xmake plugin is usually aimed at users rather than developers and documentation maintainers.

Also, executing lua scripts directly makes it easier to debug without having to put the plugin in the plugin directory each time to run it.

We just need to write a `build.lua` file to build them to webpages.

for example:

```console
xmake l build.lua -o /outputdir -s siteroot
```

we can also parse argument options, like this.

https://github.com/xmake-io/xmake/blob/master/xmake/modules/private/utils/bin2c.lua#L104